flycheck.el

```diff
@@ -2930,7 +2955,35 @@ Slots:
      in order to be able to present them to the user.
 
      See `flycheck-related-errors`."
-  buffer checker filename line column message level id group)
+  buffer checker filename start-line start-column end-line end-column region message level id group)
+
+(defun flycheck-error-calc-line-col-from-region (err)
+  (if (and (flycheck-error-region err)
+           (not (flycheck-error-start-line err)))
+      (flycheck-error-with-buffer err
+        (save-restriction
+          (save-excursion
+            (widen)
+            (goto-char (car (flycheck-error-region err)))
+            (setf (flycheck-error-start-line err) (line-number-at-pos))
+            (setf (flycheck-error-start-column err) (current-column))
+            (goto-char (cdr (flycheck-error-region err)))
+            (setf (flycheck-error-end-line err) (line-number-at-pos))
+            (setf (flycheck-error-end-column err) (current-column)))))))
+
+(defun flycheck-error-line (err)
+  (flycheck-error-calc-line-col-from-region err)
+  (flycheck-error-start-line err))
```

Comment:
Shouldn't we have accessors of this style for all 4 fields?

Reply:
I only added this to make sure I wasn't breaking the "public" API of the error struct.